### PR TITLE
Library delegates: reduce includes, remove unneeded functions and debug asserts

### PR DESCRIPTION
**Pull request by ronso0**
Just a few bycatches from #12582

### CODE REVIEW

File: src/library/previewbuttondelegate.cpp [hunk 1]
```diff
@@ -26,6 +25,23 @@ inline TrackModel* trackModel(QTableView* pTableView) {
...
+    setObjectName("LibraryPreviewButton");
+}
+
+void LibraryPreviewButton::paint(QPainter* painter) {
+    // This matches the implementation of QPushButton::paintEvent, except it
+    // does not create a new QStylePainter, and it is simpler and more
+    // direct than QWidget::render(QPainter*, ...).
+    QStyleOptionButton option;
+    initStyleOption(&option);
+    auto pStyle = style();
```
Comment (daschuer) [inline src/library/previewbuttondelegate.cpp:39]: 'auto pStyle' can be declared as 'auto *pStyle' [readability-qualified-auto]
```suggestion
    auto* pStyle = style();
```

**Review by daschuer — CHANGES_REQUESTED** [review]

File: src/library/previewbuttondelegate.cpp (hunk 1 above)
Comment (ronso0) [reply src/library/previewbuttondelegate.cpp]: done

Comment (ronso0) [reply src/library/previewbuttondelegate.cpp]: I removed another unneeded m_TableView member, it's in the base class already.

**Review by daschuer — APPROVED** [review]
LGTM, Thank you.

**Pull request merged**